src/OrchardCore.Modules/OrchardCore.Navigation/PagerShapes.cs

```diff
@@ -431,6 +407,13 @@ public Task<IHtmlContent> Pager(Shape shape, DisplayContext displayContext)
                 await shape.AddAsync(previousItem);
                 shape.Properties["FirstClass"] = PreviousClass;
             }
+            else
+            {
+                if (routeData.ContainsKey("before"))
+                {
+                    routeData.Remove("before");
```

Comment:
The route dictionary checks for you if it contains the key.

Reply:
It needs to be removed in case that we are on the first page or the last page. Here I am adding any QueryString params that might be passed and I found some places where the before or the after is passed even if it's not necessary which cause issues.

Reply:
Okay no problem, what I said is that `routeData.Remove()` checks internally if the key exists https://github.com/dotnet/aspnetcore/blob/c481a38000428c50efcbd2fc075f6084cf7a3227/src/Http/Http.Abstractions/src/Routing/RouteValueDictionary.cs#L491, so can be done in one line without having to do `ContainsKey()` line 412.